File: starters/CSharp/Bomberjam/State.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;

namespace MyBot.Bomberjam
{
    public class State
    {
        [JsonPropertyName("tiles")]
        public string Tiles { get; set; }

        [JsonPropertyName("tick")]
        public int Tick { get; set; }

        [JsonPropertyName("isFinished")]
        public bool IsFinished { get; set; }

        [JsonPropertyName("players")]
        public Dictionary<string, Player> Players { get; set; }

        [JsonPropertyName("bombs")]
        public Dictionary<string, Bomb> Bombs { get; set; }

        [JsonPropertyName("bonuses")]
        public Dictionary<string, Bonus> Bonuses { get; set; }

        [JsonPropertyName("width")]
        public int Width { get; set; }

        [JsonPropertyName("height")]
        public int Height { get; set; }

        [JsonPropertyName("suddenDeathCountdown")]
        public int SuddenDeathCountdown { get; set; }

        [JsonPropertyName("isSuddenDeathEnabled")]
        public bool IsSuddenDeathEnabled { get; set; }

        public TileKind GetTileAt(int x, int y)
        {
            if (this.IsOutOfBounds(x, y))
                return TileKind.OutOfBounds;

            var tileChar = this.Tiles[this.CoordToTileIndex(x, y)];
            return Constants.TileCharToTileKindMappings[tileChar];
        }

        private bool IsOutOfBounds(int x, int y)
        {
            return x < 0 || y < 0 || x >= this.Width || y >= this.Height;
        }

        private int CoordToTileIndex(int x, int y)
        {
            return y * this.Width + x;
        }

        public bool TryFindActiveBombAt(int x, int y, out Bomb bomb)
        {
            bomb = this.Bombs.Values.FirstOrDefault(b => b.Countdown > 0 && b.X == x && b.Y == y);
            return bomb != null;
        }

        public bool TryFindAlivePlayerAt(int x, int y, out Player player)
        {
            player = this.Players.Values.FirstOrDefault(p => p.IsAlive && p.X == x && p.Y == y);
            return player != null;
        }

        public bool TryFindDroppedBonusAt(int x, int y, out Bonus bonus)
        {
            bonus = this.Bonuses.Values.FirstOrDefault(b => b.X == x && b.Y == y);
            return bonus != null;
        }
    }
}

File: starters/Java/bomberjam/State.java
package bomberjam;

import com.google.gson.annotations.SerializedName;

import java.util.Map;

public class State {
    @SerializedName("tiles")
    private String tiles;

    @SerializedName("tick")
    private Integer tick;

    @SerializedName("isFinished")
    private Boolean isFinished;

    @SerializedName("players")
    private Map<String, Player> players;

    @SerializedName("bombs")
    private Map<String, Bomb> bombs;

    @SerializedName("bonuses")
    private Map<String, Bonus> bonuses;

    @SerializedName("width")
    private Integer width;

    @SerializedName("height")
    private Integer height;

    @SerializedName("suddenDeathCountdown")
    private Integer suddenDeathCountdown;

    @SerializedName("isSuddenDeathEnabled")
    private Boolean isSuddenDeathEnabled;

    public String getTiles() {
        return tiles;
    }

    public void setTiles(String tiles) {
        this.tiles = tiles;
    }

    public Integer getTick() {
        return tick;
    }

    public void setTick(Integer tick) {
        this.tick = tick;
    }

    public Boolean isFinished() {
        return isFinished;
    }

    public void setIsFinished(Boolean isFinished) {
        this.isFinished = isFinished;
    }

    public Map<String, Player> getPlayers() {
        return players;
    }

    public void setPlayers(Map<String, Player> players) {
        this.players = players;
    }

    public Map<String, Bomb> getBombs() {
        return bombs;
    }

    public void setBombs(Map<String, Bomb> bombs) {
        this.bombs = bombs;
    }

    public Map<String, Bonus> getBonuses() {
        return bonuses;
    }

    public void setBonuses(Map<String, Bonus> bonuses) {
        this.bonuses = bonuses;
    }

    public Integer getWidth() {
        return width;
    }

    public void setWidth(Integer width) {
        this.width = width;
    }

    public Integer getHeight() {
        return height;
    }

    public void setHeight(Integer height) {
        this.height = height;
    }

    public Integer getSuddenDeathCountdown() {
        return suddenDeathCountdown;
    }

    public void setSuddenDeathCountdown(Integer suddenDeathCountdown) {
        this.suddenDeathCountdown = suddenDeathCountdown;
    }

    public Boolean isSuddenDeathEnabled() {
        return isSuddenDeathEnabled;
    }

    public void setIsSuddenDeathEnabled(Boolean isSuddenDeathEnabled) {
        this.isSuddenDeathEnabled = isSuddenDeathEnabled;
    }

    public TileKind getTileAt(int x, int y) {
        if (this.isOutOfBounds(x, y))
            return TileKind.OUT_OF_BOUNDS;

        char tileChar = this.tiles.charAt(this.coordToTileIndex(x, y));
        return Constants.TILE_CHAR_TO_TILE_KIND_MAPPINGS.get(tileChar);
    }

    private boolean isOutOfBounds(int x, int y) {
        return x < 0 || y < 0 || x >= this.width || y >= this.height;
    }

    private int coordToTileIndex(int x, int y) {
        return y * this.width + x;
    }

    public Bomb findActiveBombAt(int x, int y) {
        for (Bomb b : this.bombs.values())
            if (b.getCountdown() > 0 && b.getX() == x && b.getY() == y)
                return b;

        return null;
    }

    public Bonus findDroppedBonusAt(int x, int y) {
        for (Bonus b : this.bonuses.values())
            if (b.getX() == x && b.getY() == y)
                return b;

        return null;
    }

    public Player findAlivePlayerAt(int x, int y) {
        for (Player p : this.players.values())
            if (p.isAlive() && p.getX() == x && p.getY() == y)
                return p;

        return null;
    }
}